src/Build/Evaluation/Expander.cs

```diff
@@ -3315,7 +3315,7 @@ private static Type GetTypeForStaticMethod(string typeName, string simpleMethodN
             private static Type GetTypeFromAssemblyUsingNamespace(string typeName)
             {
                 string baseName = typeName;
-                int assemblyNameEnd = baseName.LastIndexOf('.');
+                int assemblyNameEnd = baseName.Length;
```

Comment:
It's incidentally tested by some of the tests I enabled (couldn't find `System.Diagnostics.Process` otherwise) but an explicit test sounds reasonable.